x/autopilot/types/parser.go

```diff
@@ -27,36 +27,25 @@ type ModuleRoutingInfo interface {
 
 // Packet metadata info specific to Stakeibc (e.g. 1-click liquid staking)
 type StakeibcPacketMetadata struct {
-	Action        string `json:"action"`
-	StrideAddress string
+	Action string `json:"action"`
 }
 
 // Packet metadata info specific to Claim (e.g. airdrops for non-118 coins)
 type ClaimPacketMetadata struct {
-	StrideAddress string
 }
 
 // Validate stakeibc packet metadata fields
 // including the stride address and action type
 func (m StakeibcPacketMetadata) Validate() error {
-	_, err := sdk.AccAddressFromBech32(m.StrideAddress)
-	if err != nil {
-		return err
-	}
 	if m.Action != "LiquidStake" {
 		return errorsmod.Wrapf(ErrUnsupportedStakeibcAction, "action %s is not supported", m.Action)
 	}
 
 	return nil
 }
 
-// Validate claim packet metadata includes the stride address
+// Should we remove this struct?
```

Comment:
Circling back on this - I don't think it's super trivial to remove it.

In the main IBC callback, we have a generic `ModuleRoutingInfo` struct that we need to `switch` on to decide whether it's a stakeibc or claim packet. 

--

I think we have two options:
* (1) Keep `ClaimPacketMetadata`, but make a change to the switch statement
* (2) Change the autopilot schema to have `action` at the root level instead of the module name
```
{"autopilot": {"action": "LiquidStake", ...}}

// instead of 

{"autopilot": {"stakeibc": {"action": "LiquidStake"}}}
```